File: 94_Binary_Tree_Inorder_Traversal.js
// Definition for a binary tree node.
class TreeNode {
  constructor(val = 0, left = null, right = null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

var inorderTraversal = function (root) {
  let result = [];
  let stack = [];
  let current = root;

  while (current || stack.length > 0) {
    while (current) {
      stack.push(current);
      current = current.left; // Move to left child
    }

    current = stack.pop(); // Visit node
    result.push(current.val);

    current = current.right; // Move to right child
  }

  return result;
};

File: 94_Binary_Tree_Inorder_Traversal.py
from typing import Optional, List

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        stack = []
        current = root
        
        while current or stack:
            while current:
                stack.append(current)
                current = current.left  # Move to left child
            
            current = stack.pop()  # Visit node
            result.append(current.val)
            
            current = current.right  # Move to right child
        
        return result
